common/src/main/kotlin/org/imdc/extensions/common/DatasetExtensions.kt

```diff
@@ -73,6 +73,300 @@ object DatasetExtensions {
         return builder.build()
     }
 
+    @Suppress("unused")
+    @ScriptFunction(docBundlePrefix = "DatasetExtensions")
+    @KeywordArgs(
+        names = ["dataset", "dataset2", "columnIndex", "columnIndex2"],
+        types = [Dataset::class, Dataset::class, Int::class, Int::class],
+    )
+    fun leftJoin(args: Array<PyObject>, keywords: Array<String>): Dataset? {
+        val parsedArgs = PyArgParser.parseArgs(
+            args,
+            keywords,
+            arrayOf("dataset", "dataset2", "columnIndex", "columnIndex2"),
+            arrayOf(Dataset::class.java, Dataset::class.java, Int::class.java, Int::class.java),
+            "leftJoin",
+        )
+        val dataset = parsedArgs.requirePyObject("dataset").toJava<Dataset>()
+        val dataset2 = parsedArgs.requirePyObject("dataset2").toJava<Dataset>()
+        val column = parsedArgs.requirePyObject("columnIndex").toJava<Int>()
+        val column2 = parsedArgs.requirePyObject("columnIndex2").toJava<Int>()
+
+        val columnName = dataset.columnNames.toList()
+        val columnName2 = dataset2.columnNames.toList()
+        val combinedColumnName = columnName + columnName2
+
+        val columnType = dataset.columnTypes.toList()
+        val columnType2 = dataset2.columnTypes.toList()
+        val combinedColumnType = columnType + columnType2
+
+        val builder = DatasetBuilder.newBuilder()
+            .colNames(combinedColumnName)
+            .colTypes(combinedColumnType)
+
+        for (row in dataset.rowIndices) {
+            val listToAppend = arrayOfNulls<Any?>(combinedColumnName.size)
+            var row2: Int? = null
+
+            dataset2.rowIndices.forEachIndexed { rowIndex, _ ->
```

Comment:
Maybe try `firstNonNullOfOrNull {}` here to express what you're trying to do more succintly